### PR TITLE
Validate datasource name without calling update api

### PR DESCRIPTION
**Pull request by akash-codemonk**
- Use EditableText for datasource name

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx [hunk 1]
```diff
@@ -1,110 +1,73 @@
...
+const FormTitle = (props: FormTitleProps) => {
+  const { input } = props;
+  const params = useParams<{ datasourceId: string }>();
+  const currentDatasource:
+    | Partial<Datasource>
+    | undefined = useSelector((state: AppState) =>
+    getDatasource(state, params.datasourceId),
+  );
+  const datasources: Datasource[] = useSelector((state: AppState) =>
+    getDataSources(state),
```
Comment (hetunandu) [inline app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx:38]: No need to pass args here
```suggestion
    useSelector(getDataSources)
```

File: app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx [hunk 2]
```diff
@@ -1,110 +1,73 @@
...
+  );
+
+  const hasNameConflict = React.useCallback(
+    (name: string) => datasources.some(datasource => datasource.name === name),
```
Comment (hetunandu) [inline app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx:42]: You will also need to consider that the same datasource will have this name so also check for id.

Comment (akash-codemonk) [reply app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx:42]: Checking this in `isInvalidDatasourceName `.

Comment (hetunandu) [reply app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx:40]: This is the correct place to check it.

Comment (hetunandu) [reply app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx:40]:
```suggestion
    (name: string) => datasources.some(datasource => datasource.name === name && datasource.id !== id),
```

**Pull request merged**